wgsl/index.bs

```diff
@@ -1906,6 +1906,10 @@ memory reference</dfn> is produced.
     * store the value to any [=memory locations|memory location(s)=] of the
         [[WebGPU#buffers|WebGPU buffer]] bound to the [=originating variable=]
     * not be executed
+Statements and expressions that perform both a load and a store using the same
```

Comment:
"Statements" here is a little confusing/ambiguous. 

E.g. 
  var a: array<i32,10>;
   a[20] = a[20] + 5;  // are the a[20] the same memory reference for this purpose?
  let p = &a[25];
  *p = *p; // Are these the same memory reference for this purpose?

If not, then the wording is a little off.  Maybe insert "expression":

> ... perform both a load and a store using the same invalid memory reference expression,...

I think that works because an expression is a segment of source text.  It does split hairs that way, so another alternate would be:

> ... perform both a load and a store using the same expression instance that evaluates to an invalid memory reference expression,...

I'd also emphasize the singular qualification on statement and expression, and move the example of RMW atomics out to another sentence.

> If a statement or expression performs both a load and a store via the same expression instance that evaluates to an invalid memory reference, then the load and the store must access the same memory locations if they access memory at all.  For example, this rule applies to read-modify-write atomic operations on an invalid memory reference.

Reply:
The intent was that to handle RMW atomics the same as (eventual) += operators. So I think, these could all fall under the same restriction as I would find it confusing that `a += 5` is not the same as `a = a + 5`.